### ISSUE TITLE
Going observer doesn't grant full map vision

### ISSUE THREAD

**Issue by YakaryBovine** [AzerothWarsLR/WarcraftLegacies#52]
Typing the -obs command is meant to make you an observer and give you vision over the whole map. Right now it doesn't give you any vision. Should be easy enough to find the relevant file.

**Comment by TracyTheBoy** [comment]
I played around with this a bit and here is what I've found out.
Swapping the call to `FogModifierStart()` and the line where the factions score status is set to defeated will result in the command working.
My guess is that the call to either `RemovePlayer(Player, PLAYER_GAME_RESULT_DEFEAT);` or `SetPlayerState(Player, PLAYER_STATE_OBSERVER, 1);` in the ScoreStatus setter changes something about the Player that prevents `FogModifierStart()` from doing its job. But I don't know enough about the code or the WC3 api to say what exactly is going on.
Hope it helps.
